json_formatter.go

```diff
@@ -60,11 +60,17 @@ func (f *JSONFormatter) Format(entry *Entry) ([]byte, error) {
 		timestampFormat = DefaultTimestampFormat
 	}
 
-	if !f.DisableTimestamp {
+	if !f.DisableTimestamp && len(f.FieldMap[FieldKeyTime]) > 0 {
 		data[f.FieldMap.resolve(FieldKeyTime)] = entry.Time.Format(timestampFormat)
 	}
-	data[f.FieldMap.resolve(FieldKeyMsg)] = entry.Message
-	data[f.FieldMap.resolve(FieldKeyLevel)] = entry.Level.String()
+
+	if len(f.FieldMap[FieldKeyMsg]) > 0 {
```

Comment:
Just compare with `""`:

```
if key := f.FieldMap.resolve(FieldKeyMsg); key != "" {
  data[key] = entry.Message
}
```

Also, note that you need to use `resolve` to get the actual value for the key.